File: src/types/member.ts
export interface Member {
  id: number;
  name: string;
  phone?: string | null;
  email?: string | null;
  address?: string | null;
  image_path?: string | null;
  status: string;
  notes?: string | null;
  is_deleted: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateMemberInput {
  name: string;
  phone?: string | null;
  email?: string | null;
  address?: string | null;
  notes?: string | null;
}

File: src-tauri/src/models/member.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Member {
    pub id: i64,
    pub name: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub phone: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub cnic: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub address: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub image_path: Option<String>,
    pub status: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub notes: Option<String>,
    pub is_deleted: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Serialize, Deserialize, Debug, Clone, Default)]
pub struct CreateMemberInput {
    pub name: String,
    pub phone: Option<String>,
    pub cnic: Option<String>,
    pub address: Option<String>,
    pub notes: Option<String>,
}
